docs/CIP/CIP-XXXX/src/cardano-service.yml

```diff
@@ -0,0 +1,664 @@
+            timestamp: '2024-06-27T12:34:56Z'
+
+  schemas:
+    welcomePayload:
+      type: object
+      properties:
+        blockchains:
+          type: array
+          items:
+            type: object
+            properties:
+              name:
+                type: string
+              network:
+                type: string
+              extensions:
+                type: array
+                items:
+                  type: object
+                  properties:
+                    name:
+                      type: string
+                    config: {}
+                    default: {}
+                    switchable:
+                      type: boolean
+                    versions:
+                      type: array
+                      items:
+                        type: string
+                  required:
+                    - name
+                    - config
+                    - default
+                    - switchable
+                    - versions
+              version:
+                type: string
+            required:
+              - name
+              - network
+              - extensions
+              - version
+      required:
+        - blockchains
+    genesisPayload:
+      type: object
+      properties:
+        activeSlotsCoefficient:
+          type: number
+          format: float
+        updateQuorum:
+          type: integer
+        maxLovelaceSupply:
+          type: string
+        networkMagic:
+          type: integer
+        epochLength:
+          type: integer
+        systemStart:
+          type: integer
+        slotsPerKesPeriod:
+          type: integer
+        slotLength:
+          type: integer
+        maxKesEvolutions:
+          type: integer
+        securityParam:
+          type: integer
+      required:
+        - activeSlotsCoefficient
+        - updateQuorum
+        - maxLovelaceSupply
+        - networkMagic
+        - epochLength
+        - systemStart
+        - slotsPerKesPeriod
+        - slotLength
+        - maxKesEvolutions
+        - securityParam
+    transactionPayload:
+      type: object
+      properties:
+        outputIndex:
+          type: number
+        txHash:
+          type: string
+        transactions:
+          type: array
+          format: string
+        resolvedInputs:
+          type: array
+          format: string
+        point:
+          type: object
+          $ref: '#/components/schemas/pointPayload'
+        chain:
+          type: string
+        tip:
+          type: object
+          $ref: '#/components/schemas/pointPayload'
+    pointPayload:
+      type: object
+      properties:
+        slot:
+          type: integer
+        hash:
+          type: string
+        height:
+          type: string
+      required:
+        - hash
+    tipPayload:
+      type: object
+      properties:
+        point:
+          $ref: '#/components/schemas/pointPayload'
+      required:
+        - point
+    protocolParameterPayload:
```

Comment:
One of the goals of this API is for it to be stable over protocol upgrades where transaction contents and protocol params can change.

This current approach establishes a strong contract with the server that's specific to a particular Cardano era. We would either need to create era types or just consider serializing the protocol parameters and pass a blob to the client, similar to how we're not concerned with transaction contents. I'm leaning towards the latter since the former does not address the goals mentioned above.

Reply:
I don't have a clear winner here. I am not so sure using a blob makes it more 'stable'.

If we use a blob, the API will remain the same over time. However, declaring the details will maintain the contract created by the API between the client and the server.

If not declared, once a new era arrives, the server and the client will try to implement something not defined in the API specs. They should go and inspect the blockchain implementation and figure out the details.

Taking long-term responsibility for maintaining the client-server contract is a strategic decision.
If it is documented well, using a blob will lower the maintenance costs and keep the development cost the same.

For me, the winner is `'a blob'` by a nose.